### ISSUE TITLE
Poor performance with ScrollView's scrollY interpolation

### ISSUE THREAD

**Issue by mrousavy** [software-mansion/react-native-reanimated#1254]
## Description

I'm interpolating on the ScrollView's `scrollY` SharedValue to move my absolute positioned header around, and it has a noticeable delay (lag) when the user scrolls.

### Screenshots

[Here is a video demo](https://i.imgur.com/V16GXaU.mp4) (can't upload here because file too large)

## Steps To Reproduce

1. Create `useAnimatedScrollHandler`
2. Bind `scrollY` to `event.contentOffset.y`
3. Interpolate absolute positioned header's (above ScrollView) `translateY` to the negative `scrollY` value
4. Notice that the header doesn't move synchronized with the ScrollView, but rather lags behind

### Expected behavior

I expect the header to move synchronously with the scroll view so it looks like one piece

### Actual behavior

The header lags behind and looks like it has weight

## Snack or minimal code example

```jsx
const scrollY = useSharedValue(0);
const onScroll = useAnimatedScrollHandler({
	onScroll: (event) => {
		scrollY.value = event.contentOffset.y;
	},
});
// ...
const headerY = useDerivedValue(() => {
	return interpolate(scrollY.value, [0, HEADER_IMAGE_HEIGHT + 30], [0, -(HEADER_IMAGE_HEIGHT + 30)], Extrapolate.CLAMP);
});
const headerStyle = useAnimatedStyle(() => ({ transform: [{ translateY: headerY.value }] }));
// ...


return (
	<Reanimated.ScrollView
		scrollEventThrottle={1}
		onScroll={onScroll}
		...>
		{/* ... */}
	</Reanimated.ScrollView>

	<Reanimated.View style={[styles.imagesContainer, headerStyle]}>
		{/* children like image goes here */}
	</Reanimated.View>
	);
// ...
const styles = StyleSheet.create({
	imagesContainer: {
		position: 'absolute',
		top: 0,
		left: 0,
		height: HEADER_IMAGE_HEIGHT,
		width: SCREEN_WIDTH,
	}
});
```

## Package versions

- React: 16.13.2
- React Native: 0.63.2
- React Native Reanimated: 2.0.0-alpha.6

**Comment by terrysahaidak** [comment]
Seems like a duplicate of https://github.com/software-mansion/react-native-reanimated/issues/1242

**Comment by mrousavy** [comment]
@terrysahaidak My bad, didn't scroll through the _closed_ issues 🤦

**Comment by terrysahaidak** [comment]
Try the latest build:
https://github.com/software-mansion/react-native-reanimated/issues/1242#issuecomment-694375820

**Comment by mrousavy** [comment]
Too bad reanimated doesn't publish snapshot versions to npm. I'm going to wait for alpha 7 in the meantime, thanks. 😄

**Comment by jakub-gonet** [comment]
We provide master builds in the CI, but they're kinda broken right now

**Comment by FutureOfRussia** [comment]
Hi, I am still seeing this issue on alpha 7

**Comment by terrysahaidak** [comment]
> Hi, I am still seeing this issue on alpha 7

Could you please post reproducing code?

**Comment by FutureOfRussia** [comment]
> > Hi, I am still seeing this issue on alpha 7
> 
> Could you please post reproducing code?

In my project, I use a FlatList from react-native-gesture-handler, but replacing it with a FlatList or ScrollView from react-native not change the situation. The delays are especially noticeable with the onMomentumScroll event on iOS.

```javascript
import React from 'react'
import { View, Platform, StyleSheet } from 'react-native'
import { FlatList } from 'react-native-gesture-handler'
import Animated, {
  useAnimatedScrollHandler, useAnimatedStyle, useSharedValue, interpolate,
} from 'react-native-reanimated'
import _ from 'lodash'

const HEADER_HEIGHT = 200

export default function Test() {
  const AnimatedFlatList = Animated.createAnimatedComponent(FlatList)
  const translationY = useSharedValue(-HEADER_HEIGHT)

  const scrollHandler = useAnimatedScrollHandler({
    onScroll: (event) => {
      translationY.value = event.contentOffset.y
    },
  })

  const animatedHeaderStyle = useAnimatedStyle(() => ({
    transform: [{
      translateY: interpolate(translationY.value, Platform.OS === 'ios'
        ? [-HEADER_HEIGHT - 1, -HEADER_HEIGHT, -HEADER_HEIGHT + 1]
        : [-1, 0, 1], [0, 0, -1]),
    }],
  }))

  return (
    <View style={styles.container}>
      <Animated.View style={[styles.header, animatedHeaderStyle]}>
        {/* Header */}
      </Animated.View>
      <AnimatedFlatList
        style={styles.container}
        contentContainerStyle={styles.content}
        data={_.range(15)}
        contentInset={{ top: HEADER_HEIGHT }}
        contentOffset={{ y: -HEADER_HEIGHT }}
        renderItem={() => <View style={styles.item} />}
        keyExtractor={(_obj: any, index: number) => index.toString()}
        onScroll={scrollHandler}
        scrollEventThrottle={16}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    position: 'absolute',
    width: '100%',
    height: HEADER_HEIGHT,
    backgroundColor: 'blue',
  },
  content: {
    padding: 30,
    paddingTop: Platform.select({ android: HEADER_HEIGHT + 30 }),
  },
  item: {
    width: '100%',
    height: 120,
    backgroundColor: 'red',
    marginVertical: 15,
  },
})
```

**Comment by terrysahaidak** [comment]
Just checked it with alpha.7 and it works great.
Could make sure you installed pods again as well as a clean the build folder and run the build again?

**Comment by FutureOfRussia** [comment]
I deleted node_modules folder, reinstalled the dependencies and cleared the cache, but the problem remained.
I also use expo 39.

**Comment by terrysahaidak** [comment]
With expo the only version supported is alpha.6 which indeed has this bug.

**Comment by mrmarktyy** [comment]
Is 2.0.0.alphq.8 supported in Expo (SDK 39) ?

**Comment by SudoPlz** [comment]
I still experiences that issue in `2.2.2` (on iOS). Shouldn't that be fixed for versions above 2.0.0.alpha?

**Comment by craftzdog** [comment]
`scrollEventThrottle={16}` solved in my case
